File: Chess/src/main/java/side_server/ClientPairingThread.java
package side_server;

import game.piece.Team;
import msg.Message;

public class ClientPairingThread extends Thread {
    public SClient client;

    public ClientPairingThread(SClient client) {
        this.client = client;
    }

    @Override
    public void run() {
        while (this.client.socket.isConnected() && this.client.isWantToPair && !this.client.isPaired) {
            try {
                Server.pairingLockForTwoPair.acquire(1);
                SClient chosenPair = null;
                while (this.client.socket.isConnected() && chosenPair == null) {
                    for (SClient client : Server.clients) {
                        if (client != this.client && !client.isPaired && client.isWantToPair) {

                            chosenPair = client;
                            this.client.pair = client;
                            client.pair = this.client;
                            this.client.isWantToPair = false;
                            client.isWantToPair = false;
                            client.isPaired = true;
                            this.client.isPaired = true;

                            Message message = new Message(Message.MessageTypes.PAIRING);
                            message.content = "Eşleştin";
                            Server.sendMessage(this.client, (message));
                            Server.sendMessage(chosenPair, (message));


                            Message clientStartMessage = new Message(Message.MessageTypes.START);
                            clientStartMessage.content = Team.WHITE;
                            Message pairClientStartMessage = new Message(Message.MessageTypes.START);
                            pairClientStartMessage.content = Team.BLACK;
                            Server.sendMessage(this.client, clientStartMessage);
                            Server.sendMessage(chosenPair, pairClientStartMessage);
                            break;
                        }
                    }
                    sleep(1000);
                }
                Server.pairingLockForTwoPair.release(1);
            } catch (InterruptedException ex) {
                System.out.println("Pairing thread could not been acquired 1 permit. There is an error occurred there.");
            }
        }
    }
}

File: Chess/src/main/kotlin/hummel/side_server/ClientPairingThread.kt
package main.kotlin.hummel.side_server

import main.kotlin.hummel.game.piece.Team
import main.kotlin.hummel.msg.Message
import main.kotlin.hummel.side_server.Server.Companion.sendMessage

class ClientPairingThread(var client: SClient) : Thread() {
    override fun run() {
        while (client.socket!!.isConnected && client.isWantToPair && !client.isPaired) {
            try {
                Server.pairingLockForTwoPair.acquire(1)
                var chosenPair: SClient? = null
                while (client.socket!!.isConnected && chosenPair == null) {
                    for (client in Server.clients!!) {
						client as SClient
                        if (client != this.client && !client.isPaired && client.isWantToPair) {
                            chosenPair = client
                            this.client.pair = client
                            client.pair = this.client
                            this.client.isWantToPair = false
                            client.isWantToPair = false
                            client.isPaired = true
                            this.client.isPaired = true
                            val message = Message(Message.MessageTypes.PAIRING)
                            message.content = "Eşleştin"
                            sendMessage(this.client, message)
                            sendMessage(chosenPair, message)
                            val clientStartMessage = Message(Message.MessageTypes.START)
                            clientStartMessage.content = Team.WHITE
                            val pairClientStartMessage = Message(Message.MessageTypes.START)
                            pairClientStartMessage.content = Team.BLACK
                            sendMessage(this.client, clientStartMessage)
                            sendMessage(chosenPair, pairClientStartMessage)
                            break
                        }
                    }
                    sleep(1000)
                }
                Server.pairingLockForTwoPair.release(1)
            } catch (ex: InterruptedException) {
                println("Pairing thread could not been acquired 1 permit. There is an error occurred there.")
            }
        }
    }
}